src/BackupMetadata.cs

```diff
@@ -89,15 +89,29 @@ public object Clone()
     }
   }
 
-  public class StripedBackupSet
+  public class StripedBackupSet : BackupMetadata
   {
     public IEnumerable<BackupMetadata> StripedBackups { get; private set; }
 
     private StripedBackupSet(IEnumerable<BackupMetadata> stripedBackups)
     {
       StripedBackups = stripedBackups;
+      SetMetadata(StripedBackups.First());
     }
 
+    private void SetMetadata(BackupMetadata firstStripe)
+    {
+     CheckpointLsn = firstStripe.CheckpointLsn;
+     DatabaseBackupLsn = firstStripe.DatabaseBackupLsn;
+     DatabaseName = firstStripe.DatabaseName;
+     FirstLsn = firstStripe.FirstLsn;
+     LastLsn = firstStripe.LastLsn;
+     PhysicalDeviceName = null;
+     ServerName = firstStripe.ServerName;
+     StartTime = firstStripe.StartTime; // or is null more appropriate?
+     BackupType = firstStripe.BackupType;

```

Comment:
There is a more OOP way to do this:
We create a constructor in `BackupMetadata` that takes a `BackupMetadata` obj (like a copy constructor).
Then here, in the `StripedBackupSet` constructor, just call `base(stripedBackups.First())`

That way the properties are initialised in the class that declares them

Reply:
Will still need to set `PhysicalDeviceName=null` in the `StripedBackupSet` constructor tho - but that makes sense, it's relevant to this class